addons/csw/functions/fnc_getCarryMagazine.sqf

```diff
@@ -19,8 +19,19 @@ params ["_vehicleMag"];
 
 private _carryMag = GVAR(vehicleMagCache) get _vehicleMag;
 if (isNil "_carryMag") then {
-    private _groups = "getNumber (_x >> _vehicleMag) == 1 && {isClass (configFile >> 'CfgMagazines' >> configName _x)}" configClasses (configFile >> QGVAR(groups));
+    private _cfgMag = configFile >> "CfgMagazines";
+    private _ammoCount = getNumber (_cfgMag >> _vehicleMag >>  "count");
+    private _magazines = "getNumber (_x >> _vehicleMag) == 1 && {isClass (configFile >> 'CfgMagazines' >> configName _x)}" configClasses (configFile >> QGVAR(groups));
+
+    // try to find a mag that has the same count as the vehicleMag first, if that doesn't exist give the first element in the group array
     _carryMag = configName (_groups param [0, configNull]);
+    {
+        if (getNumber (_cfgMag >> configName _x >> "count") isEqualTo _ammoCount) then {
+            _carryMag = configName _x;
```

Comment:
```suggestion
        private _configName = configName _x;
        if (getNumber (_cfgMag >> _configName >> "count") isEqualTo _ammoCount) then {
            _carryMag = _configName;
```
No clue if doing this with this and others will make them faster or slower, though.

Reply:
Thanks for the suggestion. I think that it is slower tho, because we have to get the name every time. The if statement eliminates that in the current code.